server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -133,6 +134,130 @@ def validate_cdn
       end
     end
 
+    def compatible_cpu_families
+      # @deployment is provided by pre-filter run of find_deployment
+      rhv_hypervisors_list = @deployment.discovered_hosts
+
+      # find the fact_name_id corresponding to "processors"
+      processors_id = FactName.where(name: 'processors').first.id
+
+      # build string containing all cpu families attached to hypervisors
+      processor_concat = ''
+
+      rhv_hypervisors_list.each do |hypervisor|
+        processor_str = hypervisor.fact_values.where("fact_name_id" => processors_id).first.value
+        processor_hash = eval(processor_str)
+        processor_models = processor_hash["models"]
+        processor_concat << (processor_models.join(' ; ').downcase)
+      end
+
+      cpu_families_and_default = lcd_of_cpus(processor_concat)
+      render json: cpu_families_and_default, status: 200
+    end
+
+    def lcd_of_cpus(processor_concat)
+      # description: # finds oldest architecture, brand in rhv cluster
```

Comment:
At a minimum lcd_of_cpus should definitely be in a util class. I would probably be ok with the compatible_cpu_families staying in the controller (despite what I said in the above comment).